build(aio): migrate plunker to stackblitz

Foxandxss opened the pull request:
Stackblitz support for the AIO docs.

Should be complete, but I am sure I am missing a tiny bit or something.

ISSUES:

* There is a chunk of code at `live-example.component.ts` that deals with narrowScreens that won't work with this update but I require someone that knows more about it to fix it.
* Testing chapter has some plunkers that needs systemjs but I remember Ward wanting to remove them in a upcoming testing chapter upgrade.

Foxandxss committed:
build(aio): migrate plunker to stackblitz

mary-poppins commented:
You can preview 79edb73 at https://pr20165-79edb73.ngbuilds.io/.

EricSimons commented:
Wow, that was fast! Awesome job @Foxandxss! 💯 

On our end, we should be 100% ready to rock&roll by the end of next week - just have a few remaining pieces of func & perf improvements to finish up 👍 Specifically, we'll be rolling out:

- [ ] Hydrate dependencies/type definitions on page load to ensure the preview boots in <=4 secs (currently can hang for a _long_ time bc the browser has to download hundreds of d.ts files individually from unpkg)
  - Note: We currently do this for regular stackblitz projects automatically, but since these projects are generated on demand via the API we just need to tweak our architecture a bit to accommodate
- [ ] Deploy HyperBoot, our new in-browser dev server, to prod. This will increase bundling & boot performance by >5x
- [ ] Upgraded mobile & embed views
- [ ] Various last mile fixes
  - [ ] Enable "forking" functionality for these projects
  - [ ] Ensure "export" functionality works as intended
  - [ ] UI upgrades

Also, in regards to uptime reliability, we've intentionally designed our architecture to scale gracefully & horizontally. Specifically, we're powering the API endpoint all examples hit w/ a lambda that's decoupled from stackblitz's core app servers & the example projects themselves are stored/cached in redis. This ensures all traffic to the examples will be in-memory cache hits, avoiding any potential DB bottlenecks on the app server. Our core app servers are also load balanced & on auto-scale, should traffic ever get super crazy 👍

petebacondarwin commented:
@Foxandxss - it would be useful to rebase off the current master. There are conflicts

Foxandxss commented:
Yes, I was waiting for more merges, that will bring possibly more conflicts.

Toxicable commented:
closes #20045

petebacondarwin commented:
We are showing holding images for embedded examples (e.g. https://pr20165-79edb73.ngbuilds.io/guide/ngmodule#compile-just-in-time-jit) but the holding image is still generated from Plunker.

We need to replace all of these images across the codebase. Ideally, we should either not bother with these images (was it a performance thing?) or find some way to generate them as part of the doc-gen.

petebacondarwin commented:
@EricSimons: Is it possible to get StackBlitz to provide a URL that will result in a download of a zip file? This would allow us to remove our own zip generation and download functionality.

gkalpak commented on aio/tools/stackblitz-builder/README.md:
```diff
@@ -0,0 +1,58 @@
+
+## Appendix: Why not generating stackblitz at runtime?
+
+At AngularJS, all the stackblitz were generated a runtime. The downside is that all the example codes would
```
_stackblitz_ --> _plunkrs_ (or _live examples_) :smiley:

gkalpak commented on aio/tools/stackblitz-builder/builder.js:
```diff
@@ -8,40 +8,32 @@ var jsdom = require("jsdom");
-var fileTranslator = require('./translator/fileTranslator');
-var indexHtmlRules = require('./translator/rules/indexHtml');
-var systemjsConfigExtrasRules = require('./translator/rules/systemjsConfigExtras');
-var mainTsRules = require('./translator/rules/mainTs');
```
Woohoo 🎉 
Aren't there more SystemJS-related files that we can remove now?

Foxandxss replied:
Probably not. We will see. I will do a cleanup PR soon.

gkalpak commented on aio/tools/stackblitz-builder/builder.js:
```diff
@@ -134,12 +120,17 @@ class PlunkerBuilder {
 
       var relativeFileName = path.relative(config.basePath, fileName);
 
+      // A custom index-xxx.html file? Rename it
```
This comment doesn't seem to describe what is happening below :confused:

Foxandxss replied:
Actually it does, but I am tweaking the comment a bit.

gkalpak commented on aio/tools/stackblitz-builder/builder.js:
```diff
@@ -134,12 +120,17 @@ class PlunkerBuilder {
+      }
+
+      // A custom main.ts file? Rename it
+      if (/src\/main[-.]\w+\.ts$/.test(relativeFileName)) {
```
What a beautiful regex 😛
What about other references to the `main.X.ts` file? I remember seeing one inside an `index.html` (`System.import('...')`).

Foxandxss replied:
That shouldn't be the case anymore

gkalpak commented on aio/tools/stackblitz-builder/builder.js:
```diff
@@ -272,35 +214,32 @@ class PlunkerBuilder {
       fileName = fileName.trim();
       if (fileName.substr(0,1) == '!') {
-        return "!" + path.join(config.basePath, fileName.substr(1));
+        return "!" + fileName.substr(1);
```
`return fileName` :smiley:

gkalpak reviewed (commented):
Feels good to see all that SystemJS-specific code go :smiley:

A couple of questions/suggestions:
- In order for the ServiceWorker to route the stackblitz URLs correctly, you need to update [this RegExp][1].
- It would be nice if we could replace "stackblitz" with something generic (such as "live-example") whenever possible.

[1]: https://github.com/IdeaBlade/angular/blob/79edb73f23f391b996a2f8c4fdcf5722fc8afda8/aio/ngsw-manifest.json#L22

EricSimons commented:
As mentioned in the slack channel, we're changing the API url later today from

```
https://run.stackblitz.com/api/aio/v1
```
to

```
https://run.stackblitz.com/api/angular/v1
```
per request of the Ang Material folks. We'll add in a redirect from the aio->angular url so that nothing breaks, but the url should prob be updated to the new one before merging this PR into master 👍

petebacondarwin commented:
@Foxandxss we should really land this PR in the next few days. Do you have time to work on it or shall we bump it to someone else?

Foxandxss commented:
Ill give it love at Wed morning. Tomorrow I am out giving a course.

EricSimons commented:
New endpoint URL (https://run.stackblitz.com/api/angular/v1) is now on prod! 👍

Foxandxss commented on aio/content/examples/ngmodule/contact.1b.stackblitz.json:
```diff
@@ -0,0 +1,24 @@
+    "src/styles.css",
+    "src/index.1b.html"
+  ],
+  "main": "src/index.1b.html",
```
Why?

gkalpak replied:
Oh, it was intentional? What is `main` used for?

Foxandxss replied:
Which file to use as the "index.html". So by default it will pick the `index.html` but if you have a different one, say `index.1b.html` then you set it to be the main one. This is not stackblitz, this is how we developed the tool in the first place.

Foxandxss commented on aio/content/examples/architecture/stackblitz.json:
```diff
@@ -1,9 +1,9 @@
-  ]
+    "!**/*.[1].*"
+  ],
+  "file": "src/app/app.module.ts"
```
Stackblitz API expects a "file" so its better not to rename it locally.

Foxandxss commented on aio/content/examples/testing/1st-specs.stackblitz.json:
```diff
@@ -0,0 +1,13 @@
+    "src/1st-specs.html"
+  ],
+  "main": "src/1st-specs.html",
+  "open": "app/1st.spec.ts",
```
No, I didn't touch testing because testing has a few problems by itself that needs to be fixed with a new PR from Ward to upgrade testing to CLI. Still, I will rename that to the real deal

gkalpak replied:
So, does it mean that this example does not work anyway on stackblitz right now?

Foxandxss replied:
That is correct. Testing needs an overhaul both in testing itself and live examples

Foxandxss commented on aio/content/examples/testing/stackblitz.json:
```diff
@@ -0,0 +1,18 @@
+{
+  "description": "Heroes Test App",
+  "basePath": "src/",
```
same as above

Foxandxss commented on aio/content/examples/testing/stackblitz.json:
```diff
@@ -0,0 +1,18 @@
+    "src/index.html"
+  ],
+  "tags": ["testing"]
+}
```
same as above

Foxandxss commented on aio/tools/examples/README.md:
```diff
@@ -28,15 +28,15 @@ in any Angular application using System.js. This is the boilerplate that will be
 
 * **src/systemjs.config.js** - This is the configuration of System.js used to run the example locally.
-* **src/systemjs.config.web.js** - This configuration replaces the previous one on Plunkers.
+* **src/systemjs.config.web.js** - This configuration replaces the previous one on Stackblitz.
```
Oh. I need to update this completely (the file) because now we have CLI. I have to create a PR to update certain docs for CLI stuff.

gkalpak replied:
¯\\\_(ツ)_/¯

Foxandxss commented on aio/tools/stackblitz-builder/builder.js:
```diff
@@ -272,35 +214,32 @@ class PlunkerBuilder {
-      '!**/environments/**',
-      // AoT related files
-      '!**/aot/**/*.*',
-      '!**/*-aot.*'
```
Environments are needed to exist in the stackblitz (in plunker I change the main.ts to not load any environment).

The aot are old systemjs stuff. Our new aot projects doesn't generate those.

Foxandxss commented on aio/README.md:
```diff
@@ -4,7 +4,7 @@ Everything in this folder is part of the documentation project. This includes
 * the web site for displaying the documentation
 * the dgeni configuration for converting source files to rendered files that can be viewed in the web site.
-* the tooling for setting up examples for development; and generating plunkers and zip files from the examples.
+* the tooling for setting up examples for development; and generating stackblitz and zip files from the examples.
```
In that concrete use case, I agree.

Foxandxss commented:
Addressed some of the comments. The preview images on embedded are waiting for @wardbell to do them.

mary-poppins commented:
You can preview c2c47c3 at https://pr20165-c2c47c3.ngbuilds.io/.

gkalpak commented on aio/content/guide/ngmodule.md:
```diff
@@ -41,10 +41,10 @@ This page covers NgModules in greater depth.
+* <live-example stackblitz="minimal.0">A minimal NgModule app</live-example>
+* <live-example stackblitz="contact.1b">The first contact module</live-example>
+* <live-example stackblitz="contact.2">The revised contact module</live-example>
+* <live-example stackblitz="pre-shared.3">Just before adding SharedModule</live-example>
```
This was not addressed afaict.

Foxandxss replied:
sure, I Don't mind that.

Foxandxss replied:
Thinking twice. I do need to hard-code it. I need to search for specific files so I need to hardcode somewhere what do I need. I can have a `type="stackblitz"` attribute but I don't like that.

mary-poppins commented:
You can preview eeec1eb at https://pr20165-eeec1eb.ngbuilds.io/.

gkalpak reviewed (changes requested):
I didn't go through all the files again, but did check changes related to my previous comments.

Generally LGTM 👍 ❤️ 🎉

There are a couple of things, which are still missing (afaik) and I wouldn't ship this change without:

- In order for the ServiceWorker to route the stackblitz URLs correctly, you need to update [this RegExp][1].
- We need to update testing to work on stackblitz too.
- The placeholder images need to be updated (maybe to something generic).

<sub>It still think it would be nice if we could replace "stackblitz" with something generic (such as "live-example") whenever possible, but happy to leave it for a future PR (if at all) :smiley:</sub>

[1]: https://github.com/IdeaBlade/angular/blob/79edb73f23f391b996a2f8c4fdcf5722fc8afda8/aio/ngsw-manifest.json#L22

petebacondarwin commented:
> replace "stackblitz" with something generic
+1 (but happy for that to go in another PR)

Foxandxss commented:
Only the image is needed (waiting for @wardbell). Current testing is quite broken, this PR won't fix it so it will be fixed in a different PR.

mary-poppins commented:
You can preview 34229ea at https://pr20165-34229ea.ngbuilds.io/.

gkalpak commented:
- Service worker routing is fixed.
- Testing is already broken (on plnkr as well), so happy to defer to a different PR (already in progress: #20697).
- The image is still missing. Fine by me to get it in on a follow-up PR right after this gets merged.

Assigning to @IgorMinar to add the `merge` label when we are ready to make the switch (both from a technical and non-technical point of view :smiley:).

mary-poppins commented:
You can preview 4cd9b3b at https://pr20165-4cd9b3b.ngbuilds.io/.

gkalpak reviewed (changes requested):
The image is too big. It doesn't look nice for certain screen widths:

![image](https://user-images.githubusercontent.com/8604205/33431877-ad480440-d5de-11e7-9fe1-611cdf4a3451.png)

Also, the "Click to run" thingy looks like it is part of the demo app. I am not sure I woud realize I am supposed to click that to run the example.
<sub>(Maybe it's just me.)</sub>

Foxandxss commented:
Oh, I forgot to reduce the weight of it. And yes, maybe is biggish. Will reduce.

The play button is to make the user see that it is an image and not an "stuck" stackblitz. But I agree that it could be confusing.

petebacondarwin commented:
Really, I think we should generate these images via dgeni at doc-gen time. We could take a snapshot (using Protractor??) of each stackblitz; then overlay a "click to open" image over the top.

Foxandxss commented:
I am all in for that Pete.

gkalpak commented:
> We could take a snapshot (using Protractor??) of each stackblitz; then overlay a "click to open" image over the top.

What do you mean "of each stackblitz"?
I am generally in favor of automating stuff, but in this particular case I am a little concerned about adding an unnecessary external dependency to our build.

petebacondarwin commented:
It would be a dependency on the stackblitz server (unless we host locally for these purposes).

petebacondarwin commented:
Perhaps it could be a utility tool that is run once to generate the image, when updating an example. But then we store the images in git and don't regenerate them on the build...

petebacondarwin commented:
@Foxandxss - I think that automating the stackblitz holding image is something that will need to come later. Can you reduce the holding images as agreed so that we can get this merged in?

mary-poppins commented:
You can preview 2b60bd9 at https://pr20165-2b60bd9.ngbuilds.io/.

petebacondarwin reviewed (approved):
The placeholder image is much better.
Something to work on:
 * when you click on the image, it disappears immediately and then there is a pause (up to 3 secs) before the stackblitz fills space. Perhaps we could prevent this blank pause by making it so that the image is not hidden but instead the stackblitz has a higher z-order so that when it is ready it covers the image?

mary-poppins commented:
You can preview 79edb73 at https://pr20165-79edb73.ngbuilds.io/.
You can preview c2c47c3 at https://pr20165-c2c47c3.ngbuilds.io/.
You can preview eeec1eb at https://pr20165-eeec1eb.ngbuilds.io/.
You can preview 34229ea at https://pr20165-34229ea.ngbuilds.io/.
You can preview 4cd9b3b at https://pr20165-4cd9b3b.ngbuilds.io/.
You can preview 2b60bd9 at https://pr20165-2b60bd9.ngbuilds.io/.
You can preview 424b644 at https://pr20165-424b644.ngbuilds.io/.

mary-poppins commented:
You can preview 08ba7f7 at https://pr20165-08ba7f7.ngbuilds.io/.

mary-poppins commented:
You can preview a8deba6 at https://pr20165-a8deba6.ngbuilds.io/.

frederikprijck commented on aio/content/guide/docs-style-guide.md:
```diff
@@ -711,32 +711,32 @@ app.component.2.ts
 <live-example></live-example>
 
-1. a link to the plunker defined by the default `plnkr.json` file located in the code sample folder with the same name as the guide page.
+1. a link to the plunker defined by the default `stackblitz.json` file located in the code sample folder with the same name as the guide page.
```
`plunker` probably should be renamed to `stackblitz`.

frederikprijck commented on aio/content/guide/docs-style-guide.md:
```diff
@@ -775,12 +775,12 @@ To link to a plunker in a folder whose name is not the same as the current guide
 <h3 class="no-toc">Live Example for named plunker</h3>
 
-To link to a plunker defined by a named `plnkr.json` file, set the `plnkr` attribute. The following example links to the plunker defined by `second.plnkr.json` in the current guide's directory.
+To link to a plunker defined by a named `stackblitz.json` file, set the `stackblitz` attribute. The following example links to the plunker defined by `second.stackblitz.json` in the current guide's directory.
```
`plunker` probably should be renamed to `stackblitz`.

Foxandxss replied:
Ah, somehow I missed them. Thanks

IgorMinar reviewed (changes requested):
I tried testing this by navigating to https://pr20165-a8deba6.ngbuilds.io/tutorial/toh-pt0#final-code-review and clicking on the live demo link, which lead me to https://pr20165-a8deba6.ngbuilds.io/generated/live-examples/toh-pt0/stackblitz.html and that returned an HTTP 404 error. Is this expected? Other links I tried seem to work fine.

Also please address the unresolved comments from @gkalpak and @frederikprijck.

Foxandxss commented on aio/tools/examples/shared/boilerplate/cli/package.json:
```diff
@@ -21,8 +21,10 @@
+    "angular-in-memory-web-api": "~0.5.0",
     "core-js": "^2.4.1",
     "rxjs": "^5.5.2",
+    "web-animations-js": "^2.3.1",
```
We decided to put it when we did the complete CLI migration. Can be removed tho

IgorMinar replied:
I'd prefer it to be removed so that people don't copy&paste the code and bring in extra code that they most likely don't need.

IgorMinar commented on aio/src/app/embedded/live-example/live-example.component.ts:
```diff
@@ -5,7 +5,7 @@ import { CONTENT_URL_PREFIX } from 'app/documents/document.service';
 import { boolFromValue, getAttrs, getAttrValue } from 'app/shared/attribute-utils';
 
-const defaultPlnkrImg = 'plunker/placeholder.png';
+const defaultStackblitzImg = 'stackblitz/placeholder.png';
```
I checked out what the user experience is like with this placeholder by navigating here: https://pr20165-a8deba6.ngbuilds.io/guide/ngmodule#run-the-app

I currently see the following issues with this:

- the default placeholder and the wording above it (in the link above) suggest to try "Hello world" app, which is confusing for the reader
- the "Click to run" text looks like it's part of the output of the app and not an instruction to the reader to activate the demo
- the iframe once loaded causes is taller and wider than the placeholder image causing unfortunate reflows and resizing
- the iframe is not tall enough to display the stackblitz ui without cropping off the bottom, if the iframe was a bit taller this problem would be resolved

I think that having a less realistically looking placeholder that would make it clear that the user needs to click on the image in order to active the editor would be better. I also suggest that you make the iframe a bit taller to avoid the cropping of the stackblitz ui.

I captured a screenshot to show the current state:

![aio-stackblitz-live-demo](https://user-images.githubusercontent.com/216296/34500900-8d4a3ce2-efc1-11e7-8952-bef0c0235eeb.gif)

EricSimons replied:
Btw stackblitz uses responsive styles/func by default, so on very wide monitors (like Igor's) these embeds will default to the full UI you normally see on stackblitz.com. This can be disabled by passing along ?embed=1 which forces it to render the two-pane version, if desired 👍

IgorMinar replied:
I think it's desirable for the stackblitz UI to be responsive, I would just like to see for us to handle the transition from the placeholder image to the actual embed better.

Or alternatively, @EricSimons could you defer loading and activation of the embed until the user clicks on it a placeholder that you generate (or have an option to autoload when the iframe enters the viewport). This way, we'd always just render the iframe without the viewer having to pay the cost of loading and activating all embeds upfront. This would make the sizing and transition problems go away and would be the most elegant solution of all.

EricSimons replied:
Yup we already have this- just add ?ctl=1 and it will prompt the user to run the example. Ex: https://stackblitz.com/edit/angular?ctl=1

(ctl = click to load)

IgorMinar replied:
thanks @EricSimons

IgorMinar commented:
I also just noticed that the testing examples no longer work with stackblitz - this is because we will need a new jasmine runner template for stackblitz and this doesn't exist today.

Example: https://pr20165-a8deba6.ngbuilds.io/guide/testing#try-the-live-example-2

mary-poppins commented:
You can preview 333e4c3 at https://pr20165-333e4c3.ngbuilds.io/.

mary-poppins commented:
You can preview 4d692b9 at https://pr20165-4d692b9.ngbuilds.io/.

mary-poppins commented:
You can preview de766da at https://pr20165-de766da.ngbuilds.io/.

mary-poppins commented:
You can preview 110431b at https://pr20165-110431b.ngbuilds.io/.

mary-poppins commented:
You can preview b3b3276 at https://pr20165-b3b3276.ngbuilds.io/.

mary-poppins commented:
You can preview 483397e at https://pr20165-483397e.ngbuilds.io/.

mary-poppins commented:
You can preview 3ea674f at https://pr20165-3ea674f.ngbuilds.io/.

mary-poppins commented:
You can preview cfc85fd at https://pr20165-cfc85fd.ngbuilds.io/.

mary-poppins commented:
You can preview 00eee5e at https://pr20165-00eee5e.ngbuilds.io/.

Foxandxss commented on aio/tools/examples/example-boilerplate.js:
```diff
@@ -60,6 +60,11 @@ BOILERPLATE_PATHS.universal = [
+  ...cliRelativePath,
+  '.angular-cli.json'
+];
+
```
I normally put a test when I add a new projectType

mary-poppins commented:
You can preview 908f4b4 at https://pr20165-908f4b4.ngbuilds.io/.

wardbell commented on aio/tools/examples/shared/boilerplate/cli/package.json:
```diff
@@ -21,8 +21,10 @@
     "@angular/platform-browser": "^5.0.0",
     "@angular/platform-browser-dynamic": "^5.0.0",
     "@angular/router": "^5.0.0",
+    "angular-in-memory-web-api": "~0.5.0",
```
Some of our samples always needed in-mem. Therefore this change merely does what used to be done a different way.

IgorMinar commented:
yay! the testing examples now work! https://stackblitz.com/angular/qoxavyyvmxr (from https://pr20165-908f4b4.ngbuilds.io/guide/testing)

wardbell commented:
@IgorMinar As of the latest series of commits, all obstacles imposed by the Testing guide have been cleared. The testing stackblitz works. Other infelicities cleared.

The Testing Guide related commits will be squashed into ONE commit after reviewers have had a chance to look at the individual commits as they have been following the progress toward the end state and may be interested in the intermediate steps.

IgorMinar commented:
- [ ] replace the placeholder image with ?ctl=1 query param (see https://github.com/angular/angular/pull/20165#discussion_r159335091), this will affect /guide/ngmodule#run-the-app and any other place that uses the placeholder image
- [ ] remove the web animations polyfill from shared package.json
- [ ] live example in https://pr20165-908f4b4.ngbuilds.io/tutorial/toh-pt0#final-code-review still leads to a 404 page

If there are any more pending issues please comment here and I'll add them to the list. thanks!

mary-poppins commented:
You can preview e02ec4a at https://pr20165-e02ec4a.ngbuilds.io/.

IgorMinar commented:
@wardbell none of the issues in https://github.com/angular/angular/pull/20165#issuecomment-355447405 have been resolved. Are you still working on those?

Foxandxss commented:
@IgorMinar I will fix it as soon as I finish some other PR stuff :) AKA in 20 minutes tops

mary-poppins commented:
You can preview 325d58f at https://pr20165-325d58f.ngbuilds.io/.

IgorMinar commented:
I still see the first three problems listed: https://github.com/angular/angular/pull/20165#issuecomment-355447405

Is this PR still work in progress?

Foxandxss commented:
Yes @IgorMinar. Left you a message in the slack. The first point needs big changes. If you want them, I do it, no problem. The other two I can do easy, but I got stuck deciding on the first one.

mary-poppins commented:
You can preview a5eca24 at https://pr20165-a5eca24.ngbuilds.io/.

gkalpak commented:
This PR now needs to be updated for 53f91189a 😇

Foxandxss commented:
It will keep breaking if we don't merge it.

wardbell committed:
build(aio): upgrade sample `package.json` files to jasmine@~2.8.0

- Update tooling to support revised testing guide (PR #20697).
- Require jasmine upgrade for examples that use marble testing.
- Copy `cli/package.json` to `testing/` and add `jasmine-marbles`.
- Resolve merge conflicts created by `NgModules` guides.

wardbell commented:
Resolved the merge conflicts introduced by the recent NgModules guide changes as a by-product of updating all sample `package.json` files to use current `jasmine` packages.

Updating jasmine packages is overdue and essential when you try marble testing with `jasmine-marbles` as the new Testing guide PR #20697 does. I did this NOW because if we forget and someone tries marble testing, the error is misleading and unfathomable. I don't want anyone else have to go through what I went through to figure out what went wrong.

I also copied the `package.json` from `boilerplate/cli` to `testing/cli` and added `jasmine-marbles` to the later, again in anticipation of the Testing guide arriving at about the same time as this PR.
If I understand correctly, this step ensures that sample of type "testing" receive this `package.json` rather than the `cli` version.

`yarn serve-and-sync` runs fine and all sample E2E tests pass.

gkalpak commented:
The lockfile in `.../tools/examples/shared` has not been updated and there are (still/new?) conflicts.
And for some reason, CircleCI thinks the last commit message is all in one line (which exceeds 120 chars) 😕

gkalpak commented on aio/content/examples/service-worker-getting-started/package.json:
```diff
@@ -21,6 +21,7 @@
     "@angular/platform-browser": "^5.0.0",
```
Is it intentional that there is a `package.json` here? Aren't examples supposed to get their `package.json` from boilerplate?

gkalpak replied:
@Foxandxss confirmed this is redundant. Removed in #21629.

IgorMinar commented on aio/tools/example-zipper/customizer/package-json/testing.json:
```diff
@@ -0,0 +1,19 @@
+    { "name": "e2e", "command": "ng e2e" }
+  ],
+  "dependencies": [
+    "web-animations-js"
```
remove?

IgorMinar reviewed (changes requested):
@gkalpak can you please take over this PR and get the CI into green state with functional preview url? thanks!

gkalpak committed:
build(aio): check for obsolete `plnkr.json` and missing `main` files

Also, remove `plnkr.json` for `service-worker-getting-started` guide,
since it is not used and ServiceWorker cannot work correctly in
plnkr/stackblitz anyway (e.g. no build step to re-compute hashes).
A zipper might be useful and can be added in a subsequent PR, but it is
currently broken (e.g. no dependency on `@angular/service-worker`).

mary-poppins commented:
You can preview 95d22be at https://pr20165-95d22be.ngbuilds.io/.

mary-poppins commented:
You can preview fd76f89 at https://pr20165-fd76f89.ngbuilds.io/.

mary-poppins commented:
You can preview 56fea44 at https://pr20165-56fea44.ngbuilds.io/.

mary-poppins commented:
You can preview 110792f at https://pr20165-110792f.ngbuilds.io/.

Foxandxss commented:
@IgorMinar I fixed your issues a week ago :P

gkalpak commented:
There are still some guides that have `plnkr.json` files. I have added a check (so we catch issues with PRs that are already in progress and might use the old `plnkr.json` files) [here][1]. I am fixing them now.

[1]: https://github.com/angular/angular/pull/20165/commits/b6c586e4d71e624a1fc09e14177f5883159fad1b#diff-2ab83d06a10ee6131502ee02a9e07b35R100

gkalpak committed:
fix(aio): remove remaining `plnkr` references

mary-poppins commented:
You can preview 9ab5df5 at https://pr20165-9ab5df5.ngbuilds.io/.

mary-poppins commented:
You can preview f02c0c6 at https://pr20165-f02c0c6.ngbuilds.io/.

mary-poppins commented:
You can preview 184f63c at https://pr20165-184f63c.ngbuilds.io/.

IgorMinar commented:
one more typo in the commit message - missing ":"

gkalpak commented:
I assume (some of) the commits will be squashed, so didn't bother fixing them (for example there are `fixup!` commits and `fix Igor comments` commits :grin:).

IgorMinar commented:
@gkalpak can you please squash the fixup commit and get this ready for merging. I'd like to land this today.

FYI: @EricSimons ^

mary-poppins commented:
You can preview 2867179 at https://pr20165-2867179.ngbuilds.io/.

mary-poppins commented:
You can preview c604804 at https://pr20165-c604804.ngbuilds.io/.

mhevery commented:
http://test/OCL:183017368:BASE:183017465:1516756183390:3ec6ff57

IgorMinar commented:
This is now live. Thank you everyone who made this happen!!!  🎉 🎈 🎆
